chore: remove unused dependencies

KristjanESPERANTO opened the pull request:
This PR removes all unnecessary dependencies. They made it difficult to install the module, as several error messages appeared during `npm install`.

ralberth commented:
Excellent PR!  Thanks for taking the time to do this. I'll do you one better: this repo isn't actively developed anymore, but generally works.  Instead of depending on the specific locked, transitive dependencies in package-lock.json, I just removed the file entirely.  This way, each build picks up the latest dependency minor versions and avoids obvious CVEs, etc.  Do a 'git pull' to grab the last commit.

Thanks for caring and sending in a PR!

KristjanESPERANTO commented:
Unfortunately, removing package-lock.json didn't help. The dependencies I wanted to remove with the PR are still listed in package.json.

ralberth commented:
Crap, sorry.  Didn't read the whole PR :-(  Problem of knocking out todos on a weekend afternoon.
I removed a bunch from package.json, have a 'git pull' and LMK.

KristjanESPERANTO committed:
chore: remove duplicate mysql dependency from package.json

KristjanESPERANTO committed:
fix: correct spelling of MagicMirror²

KristjanESPERANTO committed:
chore: remove old jshint directive for node and esversion

KristjanESPERANTO committed:
docs: update section header from "Upgrade" to "Update" in README.md

KristjanESPERANTO committed:
refactor: use template literals for debug log messages

KristjanESPERANTO commented:
I have updated my PR to the current status of your repository. In addition to removing the redundant dependency, I have also addressed a few minor issues listed here on [the module check page](https://modules.magicmirror.builders/result.html#MMM-MysqlQuery-by-ralberth). No logical changes, just chores :slightly_smiling_face:
